src/requests/BaseApi.ts

```diff
@@ -0,0 +1,145 @@
+import cookiesHandler, { CookieOpts, VRCCookie } from '../VRCCookie';
+import { VRCWrapper } from '../VRCWrapper';
+import { RequestError, UserNotAuthenticated } from '../errors';
+
+/**
+ * This class is used to handle the base API requests. This class should not be used directly.
+ */
+export class BaseApi {
+    baseClass: VRCWrapper;
+
+    constructor(baseClass: VRCWrapper) {
+        this.baseClass = baseClass;
+    }
+
+    /**
+     * Function to execute the request with the given parameters.
+     * @param pathKeys  The path keys are the query parameters that are added to the path separated by a `&` symbol.
+     * @param body The body is the data that is sent to the server.
+     * @returns A promise that resolves to the data received from the server. If there is an error then it will throw an error. If there is no data then it will return an empty object.
+     * @throws {UserNotAuthenticated} If the user is not authenticated.
+     */
+    protected async executeRequest<E>({ currentRequest,
+        pathFormated,
+        queryOptions,
+        body
+    }: VRCAPI.Generics.executeRequestType): Promise<E> {
+    
+        // if all query parameters are present in the query params variable then we can replace them with our parameter from the template
+        const currentPath = this.baseClass.basePath + pathFormated;
+
+        const url: URL = new URL(`${currentPath}${queryOptions ? "?" + queryOptions.toString() : ""}`);
```

Comment:
You don't need to pass queryOptions like that, if it is empty it will not affect the URL in any way if it is a URLSearchParams type.